### ISSUE TITLE
torch to caffe convert error

### ISSUE THREAD

**Issue by Hwijune** [JDAI-CV/fast-reid#174]
```
59,60c59
<     pad: 1
<     ceil_mode: false
---
>     pad: 0
2435a2435,2444
>   name: "avgpool1"
>   type: "Pooling"
>   bottom: "relu_blob49"
>   top: "avgpool_blob1"
>   pooling_param {
>     pool: AVE
>     global_pooling: true
>   }
> }
> layer {
2438c2447
<   bottom: "relu_blob49"
---
>   bottom: "avgpool_blob1"
2449c2458
<   top: "batch_norm_blob54"
---
>   top: "output"

```
https://github.com/JDAI-CV/fast-reid/tree/master/tools/deploy
I saw the site above and converted model.

./run_inference.sh 
![image](https://user-images.githubusercontent.com/37058972/87611988-579f9080-c744-11ea-820c-fa45696e4523.png)
What should I do?

**Comment by L1aoXingyu** [comment]
Did you get the converted caffe model?

**Comment by Hwijune** [comment]
> Did you get the converted caffe model?

The previous version was 1.5. so i got error

I solved it by updating the torch version to 1.1.

![image](https://user-images.githubusercontent.com/37058972/87620158-cb4b9880-c758-11ea-904b-793ff75bcf99.png)

but i met a new error.

I can solve it soon.

appreciate of you

**Comment by gmt710** [comment]
Hello, Do we need to install caffe?

**Comment by Hwijune** [comment]
> Hello, Do we need to install caffe?

You need to add your code to "caffe_inference.py"
specify your caffe path

  #caffe_inference.py

  caffe_root = '/home/phj8498/caffe/'
  sys.path.insert(0, caffe_root + 'python')
  import caffe

**Comment by gmt710** [comment]
caffe with python3?

**Comment by Hwijune** [comment]
> caffe with python3?

ok. i'm using python3

**Comment by gmt710** [comment]
I do not use anacocnda, only compile python3 caffe.But the dependence of boost can't use. 
Do you have any suggestions to make it easier for me to do it？

**Comment by Hwijune** [comment]
> I do not use anacocnda, only compile python3 caffe.But the dependence of boost can't use.
> Do you have any suggestions to make it easier for me to do it？

Can I see the error that is currently displayed?

**Comment by gmt710** [comment]
Of course. After install boost,compiling caffe by this command "make all". Get this error:
```                
.build_release/lib/libcaffe.so: undefined reference to `boost::system::detail::generic_category_ncx()'
collect2: error: ld returned 1 exit status
```

**Comment by Hwijune** [comment]
> Of course. After install boost,compiling caffe by this command "make all". Get this error:
> 
> ```
> .build_release/lib/libcaffe.so: undefined reference to `boost::system::detail::generic_category_ncx()'
> collect2: error: ld returned 1 exit status
> ```

I have never installed caffe on anaconda. 

https://github.com/BVLC/caffe/issues/4621

How about installing original caffe? is it origin?

**Comment by gmt710** [comment]
yes, I installed the original caffe.But it doesn't work.

**Comment by gmt710** [comment]
Would you mind sharing your steps installing caffe by anaconda?
I've tried this:https://blog.csdn.net/weixin_39916966/article/details/93221564.
But cannot import caffe. I also do not clearly know where's caffe path in anaconda.

**Comment by gmt710** [comment]
Hello, about you Makefile.config?Do you install boost? opencv?and any other dependence?

**Comment by gmt710** [comment]
I think that's the problem(do not success install boost):
![image](https://user-images.githubusercontent.com/34018741/88146676-c0987400-cc2e-11ea-9772-5a707e281c6d.png)

**Comment by gmt710** [comment]
Thank you very much for your reply and guidance. I solved my problem and wrote a blog:https://blog.csdn.net/qq_35975447/article/details/107507434.

**Comment by gmt710** [comment]
I install origin caffe, occur an error:
```             
Error parsing text-format caffe.NetParameter: 60:14: Message type "caffe.PoolingParameter" has no field named "ceil_mode".
```    
I try dujianyi/caffe, but occur an error:
```             
undefined symbol: _ZN5boost6python6detail11init_moduleER11PyModuleDefPFvvE
```    
After solving the above problems, run sh ./run_inference.sh and output (do not have dilate conv)
```
Error parsing text-format caffe.NetParameter: 22:13: Message type "caffe.ConvolutionParameter" has no field named "dilation".     
```
So tired.

**Comment by Hwijune** [comment]
> 원산지 caffe를 설치하면 오류가 발생합니다.
> 
> ```
> Error parsing text-format caffe.NetParameter: 60:14: Message type "caffe.PoolingParameter" has no field named "ceil_mode".
> ```
> 
> dujianyi / caffe를 시도했지만 오류가 발생합니다.
> 
> ```
> undefined symbol: _ZN5boost6python6detail11init_moduleER11PyModuleDefPFvvE
> ```
> 
> 위의 문제를 해결 한 후 sh ./run_inference.sh를 실행하고 출력 (확장 전환이 없음)
> 
> ```
> Error parsing text-format caffe.NetParameter: 22:13: Message type "caffe.ConvolutionParameter" has no field named "dilation".     
> ```
> 
> 너무 피곤 해요

**Comment by gmt710** [comment]
I' ve solved this.I only can use BVLC/caffe.

**Comment by gmt710** [comment]
Hello, I have successfully converted the model to Caffemodel and can do inference, but the result are different from visualize_result.py.
1.I've modified prototxt file.It's same with [official deployment notes](https://github.com/JDAI-CV/fast-reid/tree/dev_face/tools/deploy).
2.Referring to [this](https://github.com/JDAI-CV/fast-reid/issues/182), delete the mean and variance processing section, but it doesn't work.
3.I've tried removing this. It can narrow the gap of Similary between official caffe_inference.py and visualize_result.py. But also can't get the same result.
```                          
ii.Add avg_pooling right place in baseline_R50.prototxt

layer {
    name: "avgpool1"
    type: "Pooling"
    bottom: "relu_blob49"
    top: "avgpool_blob1"
    pooling_param {
        pool: AVE
        global_pooling: true
    }
}
```                        

Is there anything else in the Caffemodel.prototxt that needs to be modified?
Welcome to discuss.Thanks.
